pyam/core.py

```diff
@@ -309,6 +309,31 @@ def interpolate(self, year):
         fill_values['year'] = year
         self.data = self.data.append(fill_values, ignore_index=True)
 
+    def swap_time_for_year(self, inplace=False):
+        """Convert the `time` column to `year`.
+
+        Parameters
+        ----------
+        inplace: bool, default False
+            if True, do operation inplace and return None
+
+        Raises
+        ------
+        ValueError
+            "time" is not a column of `self.data`
+        """
+        if "time" not in self.data:
+            raise ValueError("time column must be datetime to use this method")
+
+        ret = copy.deepcopy(self) if not inplace else self
+
+        ret.data["year"] = ret.data["time"].apply(lambda x: x.year)
```

Comment:
what would happen here if you have two `time` entries for the same year - if I understand this correctly, the resulting `IamDataFrame` would then have a non-unique `year` in `data` for the same model/scenario/variable/region?

This does not cause an issue in the new unit tests, but maybe it makes sense to add a sanity check here similar to https://github.com/IAMconsortium/pyam/blob/25aa7f09fb49ed64569255dd2bf260fb8e62f717/pyam/utils.py#L252

Reply:
Nice will do